lib/api.js

```diff
@@ -98,6 +98,56 @@ class API {
     return JSON.parse(body);
   }
 
+  /**
+   * Uploads the analytic documentation JSON file that describes a new private
+   * analytic to deploy.
+   *
+   * @async
+   * @param {String} docJSONPath - the path to the analytic JSON
+   * @returns {object} an object containing metadata about the posted analytic
+   * @throws {APIError} if the request was unsuccessful
+   */
+  async uploadAnalytic(docJSONPath) {
+    let formData = {
+      file: fs.createReadStream(docJSONPath),
+    };
+    let uri = this.baseURL + '/analytics';
+    let body = await requests.post(uri, this.header_, {formData: formData});
+    return JSON.parse(body).analytic;
+  }
+
+  /**
+   * Uploads the Docker image for a private analytic.
+   *
+   * The Docker image must be uploaded as a `.tar`, `.tar.gz`, or `.tar.bz`.
+   *
+   * @async
+   * @param {String} analyticId - the analytic ID
+   * @param {String} imageTarPath - the path to the image tarfile
+   * @param {String} imageType - the image computation type, 'cpu' or 'gpu'
+   * @throws {APIError} if the request was unsuccessful
+   */
+  async uploadAnalyticImage(analyticId, imageTarPath, imageType) {
+    let formData = {
+      file: fs.createReadStream(imageTarPath),
+    };
+    let params = {type: imageType.toLowerCase()};
```

Comment:
Note that I removed the type checking here. The existing pattern is just to let the API handle all type checking.